src/keygen/participant.rs

```diff
@@ -409,25 +455,30 @@ impl KeygenParticipant {
         let mut global_rid = my_decom.rid;
         // xor all the rids together. In principle, many different options for combining
         // these should be okay
+        //
+        // XXX might be better if `rid` is abstracted away, and this could be replaced
+        // by a `rid.combine` method.
         for rid in rids.iter() {
             for i in 0..32 {
                 global_rid[i] ^= rid[i];
             }
         }
         self.local_storage
             .store::<storage::GlobalRid>(self.id, global_rid);
-
+        // XXX Encapsulate this transcript generation as a method.
```

Comment:
Yes, my thinking was anywhere we use a transcript for both proving and verifying, we should have that as a method to avoid the potential misalignment of text in the transcript.